### PR TITLE
feat: setup rpcserver

### LINKED ISSUE #1: Initial skeleton setup
- JSON-RPC setup - health_check
- reth node starting
- ready for event handling

### PR DESCRIPTION
**Pull request by jac18281828**
closes #1 

This pr provides the initial skeleton setup for the sidecar project.  In this change you can find:

1. project setup
2. rpc server with health check
3. config setup via json config
4. readme and documentation
5. tests and tests passing, format and lint

Suggestions

1.  clone and build the repo
2. start and query the healthcheck

### CODE REVIEW

File: layr_rpc/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+use jsonrpsee::core::async_trait;
+use jsonrpsee::proc_macros::rpc;
+use jsonrpsee::types::ErrorObjectOwned;
+
+#[rpc(server, client)]
+pub trait Rpc {
+    /// method with connection ID.
+    #[method(name = "getBlockHeight")]
+    async fn get_block_height(&self) -> Result<usize, ErrorObjectOwned>;
```
Comment (ypatil12) [inline layr_rpc/src/lib.rs:15]: Is the idea to have this connect to reth db state to get block height?

File: sidecar/src/main.rs [hunk 2]
```diff
@@ -1,3 +1,45 @@
...
+//!
+//!
+//!
+
+// external crates
+mod config;
+use layr_rpc::run_server;
+
+#[tokio::main]
+async fn main() -> eyre::Result<()> {
```
Comment (ypatil12) [inline sidecar/src/main.rs:12]: Do we want the sidecar module to take a dependency on the JSON-RPC server? My thinking was that this module would handle the indexing of state (ie. be the RethExEx) and there is a separate json-rpc server that accesses the data store that the sidecar writes to

**Review by ypatil12 — COMMENTED** [review]
Looking solid! Had 2 high level comments

File: README.md [hunk 3]
```diff
@@ -14,6 +41,87 @@ SideCar addresses the high demand for protocol data by offering a definitive, se
 - **Real-Time Data Access**: Users can access real-time information on stakes, payment schedules, APRs, and more.
```
[the remark this answers is not in the document]
Comment (jac18281828) [reply README.md:41]: ✅

File: layr_rpc/src/lib.rs (hunk 1 above)
Comment (jac18281828) [reply layr_rpc/src/lib.rs:15]: Yes, that will be in the next pr.

Comment (jac18281828) [reply layr_rpc/src/lib.rs:15]: stubs for now

File: sidecar/src/main.rs (hunk 2 above)
Comment (jac18281828) [reply sidecar/src/main.rs:12]: yeah absolutely, Reth implementation is in progress not part of this pr, this was a getting started server setup pr.

**Review by ypatil12 — APPROVED** [review]
LGTM

**Pull request merged**